Assets/MixedRealityToolkit/_Core/Utilities/ClippingPrimitive.cs

```diff
@@ -0,0 +1,256 @@
+
+        public void AddRenderer(Renderer renderer)
+        {
+            if (renderer != null && !renderers.Contains(renderer))
+            {
+                renderers.Add(renderer);
+
+                Material material = GetMaterial(renderer, false);
+
+                if (material != null)
+                {
+                    ToggleClippingFeature(material, true);
+                }
+            }
+        }
+
+        public void RemoveRenderer(Renderer renderer)
+        {
+            if (renderers.Contains(renderer))
+            {
+                Material material = GetMaterial(renderer, false);
+
+                if (material != null)
+                {
+                    ToggleClippingFeature(material, false);
+                }
+
+                renderers.Remove(renderer);
+            }
+        }
+
+        protected void OnValidate()
+        {
+            ToggleClippingFeature(true);
+            RestoreUnassignedMaterials();
+        }
+
+        protected void OnEnable()
+        {
+            Initialize();
+            UpdateRenderers();
+            ToggleClippingFeature(true);
+        }
+
+        protected void OnDisable()
+        {
+            UpdateRenderers();
+            ToggleClippingFeature(false);
+        }
+
+        protected void Update()
+        {
+            if (Application.isPlaying || !Application.isEditor)
+            {
+                return;
+            }
+
+            Initialize();
+            UpdateRenderers();
+        }
+
+        protected void LateUpdate()
+        {
+            UpdateRenderers();
+        }
+
+        protected void OnDestroy()
+        {
+            if (renderers == null)
+            {
+                return;
+            }
+
+            for (int i = 0; i < renderers.Count; ++i)
+            {
+                Material material = GetMaterial(renderers[i]);
+
+                if (material != null)
+                {
+                    bool clippingPlaneOn;
+
+                    if (modifiedMaterials.TryGetValue(material, out clippingPlaneOn))
+                    {
+                        ToggleClippingFeature(material, clippingPlaneOn);
+                        modifiedMaterials.Remove(material);
+                    }
+                }
+            }
+
+            RestoreUnassignedMaterials();
+
+            for (int i = 0; i < allocatedMaterials.Count; ++i)
+            {
+                Destroy(allocatedMaterials[i]);
+            }
+        }
+
+        protected virtual void Initialize()
+        {
+            materialPropertyBlock = new MaterialPropertyBlock();
+            clippingSideID = Shader.PropertyToID(ClippingSideProperty);
+        }
+
+        protected virtual void UpdateRenderers()
+        {
+            if (renderers == null)
+            {
+                return;
+            }
+
+            for (int i = 0; i < renderers.Count; ++i)
+            {
+                Renderer renderer = renderers[i];
```

Comment:
There's a couple of places in each of the scripts where there are variable names that hide each other.

A simple fix is an underscore prefix for these local vars `_renderer`

Reply:
Ah, I wish Unity would remove that obsolete member. It's odd I don't get that warning locally anymore?
![renderer](https://user-images.githubusercontent.com/13305729/50565558-d1983200-0ce4-11e9-89ed-ab08bdf1520e.jpg)


I'll change them all to _renderer just in case.